elements/lisk-codec/src/codec.ts

```diff
@@ -113,6 +120,39 @@ export class Codec {
 		return (res as unknown) as T;
 	}
 
+	// For performance applications use decode() instead!
```

Comment:
I don't get this comment. `decodeJSON` and `decode` have very different use cases. Can't replace each other for performance reasons.

Reply:
It's just in case someone thinks it can just use it directly. Imagine someone is working in a custom plugin and thinks "hey I can just use my object directly with this methods decodeJSON/encodeJSON" that might be or not what they want. So it's there as a warning to call the attention of the user.

Reply:
I think it depends upon the use case if one needs JSON, it will use it, others don't.

Reply:
I think the comment does not harm. When documentation is written for lisk-codec maybe this comment can be removed. I think it doesn't hurt to leave it for now.